tests/pipeline_test.py

```diff
@@ -21,19 +21,21 @@ def int_float_to_str(x: int, y: float) -> str:
     return f"{x};{y}"
 
 
-def test_make_pipeline_sets_up_working_pipeline() -> None:
+def test_pipeline_with_callables_can_compute_single_results() -> None:
     pipeline = sl.Pipeline([int_to_float, make_int])
     assert pipeline.compute(float) == 1.5
     assert pipeline.compute(int) == 3
 
 
-def test_make_pipeline_does_not_autobind() -> None:
+def test_pipeline_does_not_autobind_types_that_can_be_default_constructed() -> None:
+    # `int` can be constructed without arguments (and returns 0). Make sure that
+    # the pipeline does not automatically bind `int` to `0`.
     pipeline = sl.Pipeline([int_to_float])
     with pytest.raises(sl.UnsatisfiedRequirement):
         pipeline.compute(float)
 
 
-def test_intermediate_computed_once() -> None:
+def test_intermediate_used_multiple_times_is_computed_only_once() -> None:
     ncall = 0
 
     def provide_int() -> int:
```

Comment:
Do you maybe want to use `unittest.mock` for counting calls?

Something like this...? Then we can use the call counting method of `Mock`.
```
from unittest.mock import Mock
MockCounter = NewType('MockCounter', Mock)
def provide_int(counter: MockCounter) -> int:
    counter()
    return 3

def test_whatever() -> None:
    mock_counter = Mock(lambda: None)
    pipeline = sl.Pipeline([int_to_float, provide_int, int_float_to_str], params={MockCounter: mock_counter})
    mock_counter.assert_called_once()
```

Reply:
We can consider this if we begin to add more complex tests. For now, the manual way seems sufficient?